lib/sql.js

```diff
@@ -156,200 +228,871 @@ SqlConnector.prototype.tableEscaped = function (model) {
-    this.toFields(model, data) +
-    ' WHERE ' + this.idColumnEscaped(model) + ' = ' + idValue;
+  var where = {};
+  where[idName] = idValue;
 
-  this.query(sql, function (err, result) {
-    if (callback) callback(err, result);
-  });
+  var updateStmt = new ParameterizedSQL('UPDATE ' + this.tableEscaped(model));
+  updateStmt.merge(this.buildFieldsForUpdate(model, data));
+  var whereStmt = this.buildWhere(model, where);
+  updateStmt.merge(whereStmt);
+  updateStmt = this.parameterize(updateStmt);
+  this.execute(updateStmt.sql, updateStmt.params, options,
+    function(err, result) {
+      if (cb) cb(err, result);
+    });
 };
 
 /**
  * Check if a model instance exists for the given id value
  * @param {String} model The model name
  * @param {*} id The id value
- * @param {Function} callback The callback function
+ * @param {Object} options Options object
+ * @param {Function} cb The callback function
  */
-SqlConnector.prototype.exists = function (model, id, callback) {
-  if (!isIdValuePresent(id, callback, true)) {
+SqlConnector.prototype.exists = function(model, id, options, cb) {
+  if (!isIdValuePresent(id, cb, true)) {
     return;
   }
-  var sql = 'SELECT 1 FROM ' +
-    this.tableEscaped(model) + ' WHERE ' +
-      this.idColumnEscaped(model) + ' = ' + this._escapeIdValue(model, id) +
-      ' LIMIT 1';
+  var idName = this.idName(model);
+  var where = {};
+  where[idName] = id;
+  var selectStmt = new ParameterizedSQL(
+    'SELECT 1 FROM ' + this.tableEscaped(model) +
+    ' WHERE ' + this.idColumnEscaped(model)
+  );
+  selectStmt.merge(this.buildWhere(model, where));
+
+  selectStmt = this.applyPagination(model, selectStmt, {
+    limit: 1,
+    offset: 0,
+    order: [idName]
+  });
+  selectStmt = this.parameterize(selectStmt);
 
-  this.query(sql, function (err, data) {
-    if (!callback) return;
+  this.execute(selectStmt.sql, selectStmt.params, options, function(err, data) {
+    if (!cb) return;
     if (err) {
-      callback(err);
+      cb(err);
     } else {
-      callback(null, data.length >= 1);
+      cb(null, data.length >= 1);
     }
   });
 };
 
 /**
- * Find a model instance by id
+ * Delete a model instance by id value
  * @param {String} model The model name
  * @param {*} id The id value
- * @param {Function} callback The callback function
+ * @param {Object} options Options object
+ * @param {Function} cb The callback function
  */
-SqlConnector.prototype.find = function find(model, id, callback) {
-  if (!isIdValuePresent(id, callback, true)) {
+SqlConnector.prototype.delete = function (model, id, options, cb) {
+  if (!isIdValuePresent(id, cb, true)) {
     return;
   }
-  var self = this;
-  var idQuery = this.idColumnEscaped(model) + ' = ' + this._escapeIdValue(model, id);
-  var sql = 'SELECT * FROM ' +
-    this.tableEscaped(model) + ' WHERE ' + idQuery + ' LIMIT 1';
+  var idName = this.idName(model);
+  var where = {};
+  where[idName] = id;
+  this.destroyAll(model, where, options, cb);
```

Comment:
FWIW and AFAIK:
 - `DataAccessObject.destroyById` calls `connector.destroyAll(model, where)` under the hood ([source](https://github.com/strongloop/loopback-datasource-juggler/blob/eb7c29826c34eb4ea90e38dd2fd03a84013d920b/lib/dao.js#L1529) and [source](https://github.com/strongloop/loopback-datasource-juggler/blob/eb7c29826c34eb4ea90e38dd2fd03a84013d920b/lib/dao.js#L1457))
 - `DataAccessObject.prototype.destroy` calls `connector.destroy(modelName, id, cb)` ([source](https://github.com/strongloop/loopback-datasource-juggler/blob/eb7c29826c34eb4ea90e38dd2fd03a84013d920b/lib/dao.js#L1931)).

It may be worth to clean this up, but I am personally ok with deferring the cleanup for a standalone pull request.